pkg/backup/resource_backupper.go

```diff
@@ -174,37 +176,59 @@ func (rb *defaultResourceBackupper) backupResource(
 	)
 
 	var namespacesToList []string
-	if resource.Namespaced {
-		namespacesToList = getNamespacesToList(rb.namespaces)
-	} else {
-		namespacesToList = []string{""}
-	}
-	for _, namespace := range namespacesToList {
-		resourceClient, err := rb.dynamicFactory.ClientForGroupVersionResource(gv, resource, namespace)
+	// If processing the Namespace resource specifically and you've specified to include only some,
+	// fetch them individually instead of a full list of all in the cluster:
+	if resource.Kind == "Namespace" &&
+		!(len(rb.namespaces.GetIncludes()) == 0 || rb.namespaces.GetIncludes()[0] == "*") {
+
+		resourceClient, err := rb.dynamicFactory.ClientForGroupVersionResource(gv, resource, "")
 		if err != nil {
 			return err
 		}
+		for _, namespace := range rb.namespaces.GetIncludes() {
+			// TODO: labelSelector relevant here?
```

Comment:
It is relevant, but GetOptions doesn't support it. You'd have to get a selector object and call Matches on it by hand.

Maybe we should just ignore this optimization for the time being?